src/battle-scene.ts

```diff
@@ -195,6 +202,52 @@ export interface InfoToggle {
   isActive(): boolean;
 }
 
+interface PokemonFaintInit {
+  preventEndure?: boolean;
+  destinyTag?: DestinyBondTag | null;
+  grudgeTag?: GrudgeTag | null;
+  source?: Pokemon;
+}
+
+interface GameOverInit {
+  isVictory?: boolean;
+  clearPhaseQueue?: boolean;
+}
+
+interface ToTitleScreenInit {
+  /** Whether to clear the phase queue before adding the {@linkcode TitlePhase}. */
+  clearPhaseQueue?: boolean;
+  /** Whether to add the {@linkcode TitlePhase} to the front of the phase queue or defer it. */
+  eager?: boolean;
+}
+
+interface ToLoginScreenInit {
+  /** Whether to show text. @default true*/
+  showText?: boolean;
+  /** Whether to add the {@linkcode LoginPhase} to the front of the phase queue or defer it. */
+  eager?: boolean;
+}
+
+interface UseMoveInit {
+  pokemon: Pokemon;
+  targets: BattlerIndex[];
+  move: PokemonMove | MoveId;
+  /** Whether to add the {@linkcode MovePhase} to the front of the phase queue or defer it. */
+  when: "eager" | "defer" | "before" | "after";
+  phaseId?: PhaseId;
+  followUp?: boolean;
+  ignorePp?: boolean;
+}
+
+//#endregion
+//#region Constants
+
+const DEBUG_RNG = false;
+
+export const startingWave = Overrides.STARTING_WAVE_OVERRIDE || 1;
```

Comment:
```suggestion
const startingWave = Overrides.STARTING_WAVE_OVERRIDE || 1;
```
Unless you changed it, this isn't used in any other files.

Reply:
I was actually thinking of moving that into `constants.ts`. Does this make sense?

Same for `DEBUG_RNG` which should then be controlled through env variables

Reply:
Well if it's only going to be used in this file then there's no need to move it. Expanding the use of these would probably be a different PR though.